pkg/util/s3client/s3-handlers.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021 The Ceph-COSI Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3client
+
+import (
+	"bytes"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+)
+
+// S3Agent wraps the s3.S3 structure to allow for wrapper methods
+type S3Agent struct {
+	Client *s3.S3
+}
+
+func NewS3Agent(accessKey, secretKey, endpoint string, debug bool) (*S3Agent, error) {
+	const cephRegion = "us-east-1"
+
+	logLevel := aws.LogOff
+	if debug {
+		logLevel = aws.LogDebug
+	}
+	sess, err := session.NewSession(
+		aws.NewConfig().
+			WithRegion(cephRegion).
+			WithCredentials(credentials.NewStaticCredentials(accessKey, secretKey, "")).
+			WithEndpoint(endpoint).
+			WithS3ForcePathStyle(true).
+			WithMaxRetries(5).
+			WithDisableSSL(true).
+			WithHTTPClient(&http.Client{
+				Timeout: time.Second * 15,
+			}).
+			WithLogLevel(logLevel),
+	)
+	if err != nil {
+		return nil, err
+	}
+	svc := s3.New(sess)
+	return &S3Agent{
+		Client: svc,
+	}, nil
+}
+
+// CreateBucket creates a bucket with the given name
+func (s *S3Agent) CreateBucketNoInfoLogging(name string) error {
+	return s.createBucket(name, false)
+}
+
+// CreateBucket creates a bucket with the given name
+func (s *S3Agent) CreateBucket(name string) error {
+	return s.createBucket(name, true)
+}
+
+func (s *S3Agent) createBucket(name string, infoLogging bool) error {
+	if infoLogging {
+		klog.Infof("creating bucket %q", name)
+	} else {
+		klog.Infof("creating bucket %q", name)
+	}
+	bucketInput := &s3.CreateBucketInput{
+		Bucket: &name,
+	}
+	_, err := s.Client.CreateBucket(bucketInput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			klog.Infof("DEBUG: after s3 call, ok=%v, aerr=%v", ok, aerr)
+			switch aerr.Code() {
+			case s3.ErrCodeBucketAlreadyExists:
+				klog.Infof("bucket %q already exists", name)
+				return nil
+			case s3.ErrCodeBucketAlreadyOwnedByYou:
+				klog.Infof("bucket %q already owned by you", name)
+				return nil
+			}
+		}
+		return errors.Wrapf(err, "failed to create bucket %q", name)
```

Comment:
I know Rook is using this `"github.com/pkg/errors"` package but to wrap an error you can just use the `fmt` package with the `%w` keyword instead.